Formula/x264.rb

```diff
@@ -52,9 +52,13 @@ class X264 < Formula
 
   depends_on "nasm" => :build
 
-  if MacOS.version <= :high_sierra
-    # Stack realignment requires newer Clang
-    # https://code.videolan.org/videolan/x264/-/commit/b5bc5d69c580429ff716bafcd43655e855c31b02
+  # Stack realignment requires newer Clang
+  # https://code.videolan.org/videolan/x264/-/commit/b5bc5d69c580429ff716bafcd43655e855c31b02
+  on_linux do
+    depends_on "gcc"
+    fails_with :clang
+  end
+  on_high_sierra :or_older do
     depends_on "gcc"
     fails_with :clang
   end
```

Comment:
Looks like this should have been macOS-only. Homebrew's compiler selection should have compiled/linked existing Linux bottle with system GCC 5 rather than current `gcc` formula.

Assuming no `gcc` formula linkage/usage in existing bottle, it might be possible to drop `on_linux` part. Otherwise, would need to rebuild Linux bottle without the unneeded `gcc` dependency.

Reply:
In a docker container, I installed `x264` and then uninstalled `gcc` and ran `brew linkage --test --strict x264`. There were no issues reported. Is that enough to show that there will be no issues if I remove the `on_linux` block?